File: daily/Arrays/1-Two-sum.go
package daily

func twoSum(nums []int, target int) []int {
	mapSum := make(map[int]int)
	for index, num := range nums {
		remain := target - num
		if value, ok := mapSum[remain]; ok {
			return []int{index, value}
		} else {
			mapSum[num] = index
		}
	}
	return []int{}
}

File: daily/Arrays/1-Two-sum.py
from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        map_sum = {}
        for index, num in enumerate(nums):
            remain = target - num
            if remain in map_sum:
                return [map_sum[remain], index]
            map_sum[num] = index
        return []
